components/chart/editModule/index.tsx

```diff
@@ -0,0 +1,282 @@
+  z-index: 1050;
+
+  .header {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    direction: ${({ lang }) =>
+    lang === null || lang === 'en' ? 'ltr' : 'rtl'};
+    .title {
+      margin-top: 10px;
+      display: flex;
+      font-weight: bold;
+      align-items: center;
+
+      .linkIcon {
+        width: 20px;
+        height: 20px;
+        cursor: pointer;
+        margin-right: 10px;
+      }
+    }
+
+    .closeIcon {
+      width: 40px;
+      height: 40px;
+      cursor: pointer;
+    }
+  }
+
+  &.show {
+    transform: translateX(0);
+    overflow-y: auto;
+    max-height: 100%;
+    padding-bottom: 10px;
+  }
+`;
+const SuccessToast = styled.div`
+  position: fixed;
+  bottom: 20px;
+  left: 64%;
+  transform: translateX(-50%);
+  background-color: #4caf50; 
+  color: #fff;
+  padding: 10px 20px;
+  border-radius: 5px;
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.5);
+  z-index: 9999;
+`;
+
+const ErrorToast = styled.div`
+  position: fixed;
+  bottom: 20px;
+  left: 50%;
+  transform: translateX(-50%);
+  background-color: #ff6347;
+  color: #fff;
+  padding: 10px 20px;
+  border-radius: 5px;
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
+  z-index: 9999;
+`;
+
+const ToastMessage = styled.span`
+  font-size: 14px;
+  font-weight: bold;
+`;
+const Button = styled.button`
+  display: block;
+  width: 85%;
+  height: 40px;
+  padding: 10px;
+  font-weight: bold;
+  font-size: 16px;
+  margin: 10px 0;
+  background-color: #699041;
+  border: solid 1px var(--bc-secondary-light-hover);
+  border-radius: 20px;
+  color: #fff;
+  cursor: pointer;
+  
+  &:hover {
+    background-color: #5a7736; /* Brown hover color */
+  }
+`;
+const EditDrawer: React.FC<PropsWithChildren<DrawerProps>> = ({ open, lang, onClose, module, children }) => {
+  const drawerRef = useRef(null);
+  const editPanelRef = useRef(null);
+  const [successMessage, setSuccessMessage] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const { data, loading } = useContentInstancesQuery(module?.id);
+  const { data: modelsResult, loading: modelsLoading } =
+    useModelsQuery(TEMPLATE_CONTEXT_ID);
+  const [
+    updateContentInstanceFieldValues,
+    {
+      data: updateContentInstanceFieldValuesCallData,
+      error: updateContentInstanceFieldValuesCallError,
+      loading: updateContentInstanceFieldValuesLoading,
+    },
+  ] = useUpdateContentInstanceFieldValuesMutation();
+
+  const targetModel = useMemo(() => {
+    if (modelsResult?.viewTypes) {
+      return modelsResult.viewTypes.find(viewType => viewType.Id === HYCHART_VIEW_TYPE_ID);
+    }
+    return undefined;
+  }, [modelsResult, module?.id]);
+
+  const { control, handleSubmit, formState: { errors }, register, setValue, getValues } = useForm<any>();
+  useEffect(() => {
+    if (targetModel && data?.contentInstances.length > 0) {
+      targetModel?.ViewFields.reduce(
```

Comment:
Usually, we are using reduce to group different items to get a final object, but here you are not using the result, so I think forEach is more logical

I used that in BC-portal because I needed to build up formInitialValue object